### DISCUSSION TITLE
Possible topics for Dev Call - 3Dec2023

### DISCUSSION THREAD

**Discussion by richb-hanover** [PRQL/prql#3882] [General]
I'm on the Discord channel now, but no one else is there. I had planned to say these things:

- I really appreciate people's attention to the Dev Container, and to tweaking the Playground. It makes it pretty straightforward for me to play around the edges
 
- I've been fooling around with `group`. I have hit (and documented) a few bugs regarding that transform, so my work there is stopped. What's the most useful way for me to proceed? More bug reports showing various aspects (of what may be the same problem)? Wait for progress in the compiler?

- What are the big issues for PRQL for the next few weeks. I have no desire to be the guy who's piping up with little problems while the core team is wrestling with big concepts?

Thanks.

@max-sixty @aljazerzen

**Comment by max-sixty** [comment]
I commented on Discord — sorry not to say here — I couldn't make today. Lmk if it went ahead and there was anything to follow up on. Otherwise look forward to speaking in a couple of weeks.

**Comment by aljazerzen** [comment]
> I've been fooling around with group. I have hit (and documented) a few bugs regarding that transform, so my work there is stopped. What's the most useful way for me to proceed? More bug reports showing various aspects (of what may be the same problem)? Wait for progress in the compiler?

Although I dislike bugs (because they have to be fixed), they are very valuable: every fixed bug adds a test case into our suite, so all future versions of the compiler will not contain the same bug. When we fix all known bugs, we start hitting the problem that we **don't know** what's not working. One can try to construct weird test cases and hope to get a bug, but it's much easier if we have a backlog issues.

So yes: please file more bugs. Even if it might be a duplicate. Even if it might be a "bad program".

The compiler will eventually catch up, I'm focusing on "under the hood" stuff right now.

**Reply by richb-hanover** [reply]
Thanks. I didn't want to be accused of "piling on". (I think I only have one more report, likely related, but wanted to know how useful that might be. I'll post it in a couple days.)
